src/db/mod.rs

```diff
@@ -232,14 +275,41 @@ impl Database {
             .await
     }
 
-    /// Creates a new collection in the database with the given `name` and `options`.
-    ///
-    /// Note that MongoDB creates collections implicitly when data is inserted, so this method is
-    /// not needed if no special options are required.
-    pub async fn create_collection(
+    /// Gets the names of the collections in the database using the provided `ClientSession`.
+    pub async fn list_collection_names_with_session(
+        &self,
+        filter: impl Into<Option<Document>>,
+        session: &mut ClientSession,
+    ) -> Result<Vec<String>> {
+        let list_collections =
+            ListCollections::new(self.name().to_string(), filter.into(), true, None);
+        let mut cursor: SessionCursor<Document> = self
```

Comment:
Given that `SessionCursorHandle` implements `Stream`, we should be able to write a generic version of this to avoid duplicating the code.